src/main/java/org/folio/circulation/resources/CheckInByBarcodeResource.java

```diff
@@ -112,7 +116,8 @@ private void checkin(RoutingContext routingContext) {
         processAdapter::getAddressType, CheckInProcessRecords::withHighestPriorityFulfillableRequest))
       .thenComposeAsync(updateItemResult -> updateItemResult.combineAfter(
         processAdapter::updateLoan, CheckInProcessRecords::withLoan))
-      .thenApply(updateItemResult -> updateItemResult.next(processAdapter::sendCheckInPatronNotice))
+      .thenComposeAsync(updateItemResult -> updateItemResult.after(
+        patronActionSessionService::saveCheckInSessionRecord))
```

Comment:
Does the check in by barcode endpoint have permissions to make this request to save the session?

Reply:
the permission has been added